app/src/main/java/com/quran/labs/androidquran/ui/TranslationManagerActivity.java

```diff
@@ -330,6 +356,86 @@ private void removeItem(final TranslationRowData translationRowData) {
     builder.show();
   }
 
+  private List<TranslationItem> sortedDownloadedItems() {
+    final ArrayList<TranslationItem> result = new ArrayList<>();
+    for (TranslationItem item : allItems) {
+      if (item.exists()) result.add(item);
+    }
+    Collections.sort(result, new TranslationItemDisplaySort ());
+    return result;
+  }
+
+  private void rankDownItem(TranslationRowData targetRow) {
+    TranslationItem targetItem = (TranslationItem) targetRow;
+    List<TranslationItem> sortedDownloads = sortedDownloadedItems();
+    if (sortedDownloads.indexOf(targetItem) + 1 < sortedDownloads.size()) { // ignore last item in list
+      TranslationItem updatedItem = targetItem.withDisplayOrder(targetItem.getDisplayOrder() + 1);
+      ArrayList<TranslationItem> toUpdate = new ArrayList<>();
```

Comment:
question - why do we have to do this? why not assume that the result from `sortDownloadedItems()` is the sort ordering - so basically when you choose item 3 and move up, you just swap positions 2 and 3 without having to do these loops.

Reply:
Again, for what was mentioned previously that there may be gaps when an item is deleted.  Alternatively, shifting logic could be moved to download routine... but that seems a lot trickier at first glance.